lib/mpi/responses/profile_parser.rb

```diff
@@ -222,6 +227,10 @@ def log_inactive_mhv_ids(mhv_ids, active_mhv_ids)
         end
       end
 
+      def validate_dob(dob, icn)
+        Rails.logger.warn 'MPI::Response.parse_dob failed', { dob: dob, icn: icn } if dob.nil? || dob.length != 8
```

Comment:
To validate the date, instead you should try: `Date.iso8601(dob)`, that will give you errors if the full date is not defined, if it is nil, and if it is not arranged in the way we expect. It's prob a more robust validation

Reply:
So looks something like

```
Date.iso8601(dob)
rescue Date::Error
  Rails.logger.warn 'MPI::Response...'
end
```